apiserver/watcher.go

```diff
@@ -86,46 +60,156 @@ type SrvAllWatcher struct {
 	watcher multiwatcher.Watcher
 }
 
+// NewAllWatcherV1 is a wrapper that creates a V1 AllWatcher API.
+func NewAllWatcherV1(context facade.Context) (facade.Facade, error) {
+	api, err := newAllWatcher(context)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	return &SrvAllWatcherV1{api}, nil
+}
+
+// Next will return the current state of everything on the first call
+// and subsequent calls will
+func (aw *SrvAllWatcherV1) Next() (params.AllWatcherNextResults, error) {
+	deltas, err := aw.watcher.Next()
+	return params.AllWatcherNextResults{
+		Deltas: translate(newAllWatcherDeltaTranslaterV1(), deltas),
```

Comment:
should the aw receiver have a deletaTranslator attribute constructed in the NewAllWatcher to avoid calling newAllWatcherDeltaTranslaterV1() every time Next() is called?

Reply:
Same with the other one